processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -555,19 +591,32 @@ private File makeIndexFiles(
       dimConversions.add(Maps.<String, IntBuffer>newHashMap());
     }
 
+    Map<String, Integer> nullStringIndexMap = new LinkedHashMap<>();
```

Comment:
It doesn't look like this needs to be Linked, if not please just use a regular HashMap